### PR TITLE
RAC-277: Tracking progress of a job

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

RAC-285: https://github.com/akeneo/pim-community-dev/pull/12993
RAC-216 / RAC-177 / RAC-280 : https://github.com/akeneo/pim-community-dev/pull/13053
RAC-281 / RAC-282 / RAC-289 / RAC-291: https://github.com/akeneo/pim-community-dev/pull/13015
RAC-214: https://github.com/akeneo/pim-community-dev/pull/13067

Todo :
- [x] Status translation in grid
- [x] Add commit to tag storybook
- [x] Add Changelog 
- [x] Add forwardRef and supports ...rest props in component ProgressBar (cf https://github.com/akeneo/pim-community-dev/pull/13162)
- [x] Remove warnings count badge in process tracker grid
- [x] Change warning count value by "-" when equal to 0

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Done
| Added legacy Behats               | -
| Added acceptance tests            | Done
| Added integration tests           | Done
| Changelog updated                 | Done
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Done
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilyProductsTasklet.php [hunk 1]
```diff
@@ -182,19 +192,58 @@ private function saveProducts(array $products): void
...
     /**
-     * @param FamilyInterface $family
+     * Quentin raised some concerns regarding the ability to rewind the familyReader (which can be seen as a cursor)
+     * in a shared environment.
      *
-     * @return CursorInterface
+     * @throws \Exception
      */
-    private function getProductsForFamily(FamilyInterface $family): CursorInterface
+    private function computeProductsToProcess(): int
```
Comment (juliensnz) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilyProductsTasklet.php:213]: did we measured the time it takes on big files?

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilyRootProductModelsTasklet.php [hunk 2]
```diff
@@ -187,18 +197,60 @@ private function saveProductsModel(array $productModels): void
...
     /**
-     * @param FamilyInterface $family
+     * Quentin raised some concerns regarding the ability to rewind the familyReader (which can be seen as a cursor)
+     * in a shared environment.
      *
-     * @return CursorInterface
+     * @throws \Exception
      */
-    private function getRootProductModelsForFamily(FamilyInterface $family): CursorInterface
+    private function computeRootProductModelsToProcess(): int
```
Comment (juliensnz) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilyRootProductModelsTasklet.php:210]: same here

File: src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilySubProductModelsTasklet.php [hunk 3]
```diff
@@ -164,6 +169,57 @@ public function initialize()
...
+        return $this->computeSubProductModelsToProcess();
+    }
+
+    /**
+     * Quentin raised some concerns regarding the ability to rewind the familyReader (which can be seen as a cursor)
+     * in a shared environment.
+     *
+     * @throws \Exception
+     */
+    private function computeSubProductModelsToProcess(): int
```
Comment (juliensnz) [inline src/Akeneo/Pim/Enrichment/Component/Product/Connector/Job/ComputeDataRelatedToFamilySubProductModelsTasklet.php:183]: same

**Review by juliensnz — APPROVED** [review]

**Comment by juliensnz** [comment]
My perf remarks are more to make sure it's ok. As long as it's less than 2/3% of the total job time, it's OK for me

**Review by ValentinMumble — APPROVED** [review]

**Pull request merged**